src/field_theory/cardinality.lean

```diff
@@ -69,3 +70,16 @@ begin
     simpa [mv_polynomial.monomial_eq_monomial_iff, finsupp.single_eq_single_iff] using h },
   { simpa using @mv_polynomial.cardinal_mk_le_max α (ulift.{u} ℚ) _ }
 end
+
+/-- There is a field structure on type if its cardinality is a prime power. -/
+lemma field.nonempty_iff {α : Type u} : nonempty (field α) ↔ is_prime_pow (#α) :=
+begin
+  rw cardinal.is_prime_pow_iff,
+  by_cases ω ≤ (#α),
```

Comment:
It's probably nicer to use
```lean
cases le_or_lt ω (#a)
```

Reply:
in this case, I want the `¬w ≤ #a` so that `simp` can do stuff like `or_false`